examples/token_update_metadata/main.go

```diff
@@ -0,0 +1,173 @@
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/hashgraph/hedera-sdk-go/v2"
+)
+
+func main() {
+	var client *hedera.Client
+	var err error
+
+	// Retrieving network type from environment variable HEDERA_NETWORK
+	client, err = hedera.ClientForName(os.Getenv("HEDERA_NETWORK"))
+	if err != nil {
+		panic(fmt.Sprintf("%v : error creating client", err))
+	}
+	// Retrieving operator ID from environment variable OPERATOR_ID
+	operatorAccountID, err := hedera.AccountIDFromString(os.Getenv("OPERATOR_ID"))
+	if err != nil {
+		panic(fmt.Sprintf("%v : error converting string to AccountID", err))
+	}
+	// Retrieving operator key from environment variable OPERATOR_KEY
+	operatorKey, err := hedera.PrivateKeyFromString(os.Getenv("OPERATOR_KEY"))
+	if err != nil {
+		panic(fmt.Sprintf("%v : error converting string to PrivateKey", err))
+	}
+	// Setting the client operator ID and key
+	client.SetOperator(operatorAccountID, operatorKey)
+
+	updateMutableTokenMetadata(client)
+
+	updateImmutableTokenMetadata(client)
+}
+
+func updateMutableTokenMetadata(client *hedera.Client) {
+
+	adminKey, err := hedera.PrivateKeyGenerateEd25519()
+	if err != nil {
+		panic(fmt.Sprintf("%v : error creating admin key", err))
+	}
+	fmt.Println("create admin key: ", adminKey.String())
+
+	var initialMetadata = []byte{1, 2, 3}
+	var newMetadata = []byte{3, 4, 5, 6}
+
+	tx, err := hedera.NewTokenCreateTransaction().
+		SetTokenName("ffff").
+		SetTokenSymbol("F").
+		SetDecimals(3).
+		SetTokenType(hedera.TokenTypeFungibleCommon). // The same flow can be executed with a TokenTypeNonFungibleUnique (i.e. HIP-765)
```

Comment:
The example could accept the token type like `-ft`, `-nft` and here you can just put in the token type after the token type is resolved.